base/mathconstants.jl

```diff
@@ -89,6 +89,7 @@ catalan
 # loop over types to prevent ambiguities for ^(::Number, x)
 for T in (AbstractIrrational, Rational, Integer, Number, Complex)
     Base.:^(::Irrational{:ℯ}, x::T) = exp(x)
+    Base.:^(::Irrational{:ℯ}, x::AbstractMatrix{T}) = exp(x)
```

Comment:
I don't know, wouldn't that still allow for the possibilities of `^(::Number, x)`?  I would have thought it can create an ambiguity if we have the fallback, even if we have the specific methods.

Reply:
In that case, maybe just add `AbstractMatrix` to the thing that `T` loops over?

Reply:
This should be defined in LinearAlgebra.

Reply:
I didn't want to add `AbstractMatrix` as `T` to the loop because I thought it would be safer (in regard to method ambiguities) to ensure that only matrices with these element types have exponentials defined for them.  Having said that, I just realized that this should have read `AbstractMatrix{<:T}` rather than what I have here.

Let me know whether you think this should indeed be fore general `AbstractMatrix` and I'll move it to `LinearAlgebra`.

Reply:
Since `exp(::AbstractMatrix)` is defined in `LinearAlgebra` this should too.